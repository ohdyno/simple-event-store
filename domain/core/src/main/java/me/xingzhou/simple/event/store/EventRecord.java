package me.xingzhou.simple.event.store;

import me.xingzhou.simple.event.store.storage.StoredRecord;

public record EventRecord(Event event, RecordDetails details) {
    public static EventRecord extract(StoredRecord record, Event event) {
        return new EventRecord(
                event, new RecordDetails(record.streamName(), record.version(), record.id(), record.insertedOn()));
    }
}
