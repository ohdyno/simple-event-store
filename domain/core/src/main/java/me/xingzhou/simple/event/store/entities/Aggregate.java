package me.xingzhou.simple.event.store.entities;

import me.xingzhou.simple.event.store.ids.StreamName;
import me.xingzhou.simple.event.store.ids.Version;

public interface Aggregate extends EventSourceEntity {
    void setVersion(Version version);

    StreamName streamName();

    Version version();
}
