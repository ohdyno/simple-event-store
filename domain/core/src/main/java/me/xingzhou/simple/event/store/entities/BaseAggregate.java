package me.xingzhou.simple.event.store.entities;

import me.xingzhou.simple.event.store.ids.Version;
import me.xingzhou.simple.event.store.storage.EventStorage;

public abstract class BaseAggregate implements Aggregate {
    private long version = EventStorage.Constants.Versions.UNDEFINED_STREAM;

    @Override
    public boolean isDefined() {
        return version != EventStorage.Constants.Versions.UNDEFINED_STREAM;
    }

    @Override
    public void setVersion(Version version) {
        this.version = version.value();
    }

    @Override
    public Version version() {
        return new Version(version);
    }
}
