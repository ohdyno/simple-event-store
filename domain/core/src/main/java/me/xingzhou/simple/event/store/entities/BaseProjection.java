package me.xingzhou.simple.event.store.entities;

import java.time.Instant;
import me.xingzhou.simple.event.store.ids.RecordId;
import me.xingzhou.simple.event.store.storage.EventStorage;

public abstract class BaseProjection implements Projection {
    private long lastRecordId = EventStorage.Constants.Ids.UNDEFINED;
    private Instant lastUpdatedOn = EventStorage.Constants.InsertedOnTimestamps.NEVER;

    @Override
    public boolean isDefined() {
        return lastUpdatedOn != EventStorage.Constants.InsertedOnTimestamps.NEVER;
    }

    @Override
    public RecordId lastRecordId() {
        return new RecordId(lastRecordId);
    }

    @Override
    public Instant lastUpdatedOn() {
        return lastUpdatedOn;
    }

    @Override
    public void setLastRecordId(RecordId id) {
        this.lastRecordId = id.id();
    }

    @Override
    public void setLastUpdatedOn(Instant instant) {
        this.lastUpdatedOn = instant;
    }
}
