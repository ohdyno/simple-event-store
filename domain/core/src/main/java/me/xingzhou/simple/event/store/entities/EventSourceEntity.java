package me.xingzhou.simple.event.store.entities;

public interface EventSourceEntity {
    String APPLY_METHOD_NAME = "apply";

    default boolean isDefined() {
        return false;
    }
}
