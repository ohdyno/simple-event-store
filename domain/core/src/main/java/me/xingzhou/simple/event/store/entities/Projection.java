package me.xingzhou.simple.event.store.entities;

import java.time.Instant;
import java.util.Collections;
import java.util.List;
import me.xingzhou.simple.event.store.ids.RecordId;
import me.xingzhou.simple.event.store.ids.StreamName;

public interface Projection extends EventSourceEntity {
    RecordId lastRecordId();

    Instant lastUpdatedOn();

    void setLastRecordId(RecordId id);

    void setLastUpdatedOn(Instant instant);

    default List<StreamName> streamNames() {
        return Collections.emptyList();
    }
}
