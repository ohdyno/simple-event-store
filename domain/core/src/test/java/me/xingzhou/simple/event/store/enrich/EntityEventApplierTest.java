package me.xingzhou.simple.event.store.enrich;

import static org.assertj.core.api.Assertions.assertThat;
import static org.junit.jupiter.api.Assertions.assertAll;
import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;

import com.github.valfirst.slf4jtest.LoggingEvent;
import com.github.valfirst.slf4jtest.TestLogger;
import com.github.valfirst.slf4jtest.TestLoggerFactory;
import java.time.Instant;
import java.util.stream.Collectors;
import me.xingzhou.simple.event.store.Event;
import me.xingzhou.simple.event.store.RecordDetails;
import me.xingzhou.simple.event.store.entities.EventSourceEntity;
import me.xingzhou.simple.event.store.events.TestEventTypeConverter;
import org.approvaltests.Approvals;
import org.approvaltests.core.Options;
import org.approvaltests.scrubbers.DateScrubber;
import org.junit.jupiter.api.Test;

public class EntityEventApplierTest {

    private static final TestLogger logger = TestLoggerFactory.getTestLogger(EntityEventApplier.class);

    @Test
    void applyEvent() {
        var event = new HierarchyEventWithInterfaces();
        var record = new EventRecord(event, new RecordDetails("stream-name", 0, 0, Instant.now()));
        var subject = new EntityEventApplier(new EventTypesExtractor(new TestEventTypeConverter()));

        assertAll(
                "apply hierarchical event to different recorders",
                () -> assertDoesNotThrow(
                        () -> {
                            var recorder = new EventAppliedRecorder();
                            subject.apply(record, recorder);
                            var eventTypeName = Event.class.getSimpleName();
                            assertThat(recorder.eventTypeApplied())
                                    .as(eventTypeName)
                                    .isTrue();
                        },
                        "apply(Event)"),
                () -> assertDoesNotThrow(
                        () -> {
                            var recorder = new BaseHierarchyEventAppliedRecorder();
                            subject.apply(record, recorder);
                            var eventTypeName = BaseHierarchyEvent.class.getSimpleName();
                            assertThat(recorder.eventTypeApplied())
                                    .as(eventTypeName)
                                    .isTrue();
                        },
                        "apply(BaseHierarchyEvent)"),
                () -> assertDoesNotThrow(
                        () -> {
                            var recorder = new TestEventInterfaceAEventAppliedRecorder();
                            subject.apply(record, recorder);
                            var eventTypeName = TestEventInterfaceA.class.getSimpleName();
                            assertThat(recorder.eventTypeApplied())
                                    .as(eventTypeName)
                                    .isTrue();
                        },
                        "apply(TestEventInterfaceA)"),
                () -> assertDoesNotThrow(
                        () -> {
                            var recorder = new TestEventInterfaceBEventAppliedRecorder();
                            subject.apply(record, recorder);
                            var eventTypeName = TestEventInterfaceB.class.getSimpleName();
                            assertThat(recorder.eventTypeApplied())
                                    .as(eventTypeName)
                                    .isTrue();
                        },
                        "apply(TestEventInterfaceB)"),
                () -> assertDoesNotThrow(
                        () -> {
                            var recorder = new HierarchyEventWithInterfacesEventAppliedRecorder();
                            subject.apply(record, recorder);
                            var eventTypeName = HierarchyEventWithInterfaces.class.getSimpleName();
                            assertThat(recorder.eventTypeApplied())
                                    .as(eventTypeName)
                                    .isTrue();
                        },
                        "apply(HierarchyEventWithInterfaces)"),
                () -> assertDoesNotThrow(
                        () -> {
                            var recorder = new SubRecorder();
                            subject.apply(record, recorder);
                            var eventTypeName = Event.class.getSimpleName();
                            assertThat(recorder.eventTypeApplied())
                                    .as(eventTypeName)
                                    .isTrue();
                        },
                        "apply(Event) from super class"),
                () -> assertDoesNotThrow(
                        () -> {
                            var recorder = new EventWithRecordDetailsRecorder();
                            subject.apply(record, recorder);
                            var eventTypeName = Event.class.getSimpleName();
                            assertThat(recorder.eventTypeApplied())
                                    .as(eventTypeName)
                                    .isTrue();
                            assertThat(recorder.recordedDetails()).isEqualTo(record.details());
                        },
                        "apply(<? extends Event>, RecordDetails)"));

        var loggedEvents = logger.getAllLoggingEvents().stream()
                .map(LoggingEvent::toString)
                .collect(Collectors.joining("\n"));
        Approvals.verify(
                loggedEvents, new Options().withScrubber(DateScrubber.getScrubberFor("2025-04-20T04:21:48.191772Z")));
    }

    public static class BaseHierarchyEvent implements Event {
        @Override
        public String toString() {
            return "BaseHierarchyEvent{}";
        }
    }

    public static class BaseHierarchyEventAppliedRecorder extends Recorder {
        public void apply(BaseHierarchyEvent event) {
            eventTypeApplied = true;
        }

        @Override
        public String toString() {
            return "BaseHierarchyEventAppliedRecorder{" + "eventTypeApplied=" + eventTypeApplied + '}';
        }
    }

    public static class EventAppliedRecorder extends Recorder {
        public void apply(Event event) {
            eventTypeApplied = true;
        }

        @Override
        public String toString() {
            return "EventAppliedRecorder{}";
        }
    }

    public static class EventWithRecordDetailsRecorder extends Recorder {
        private RecordDetails recordedDetails = null;

        public void apply(Event event, RecordDetails recordDetails) {
            eventTypeApplied = true;
            recordedDetails = recordDetails;
        }

        @Override
        public String toString() {
            return "EventWithRecordDetailsRecorder{" + "recordedDetails=" + recordedDetails + ", eventTypeApplied="
                    + eventTypeApplied + '}';
        }

        private RecordDetails recordedDetails() {
            return recordedDetails;
        }
    }

    public static class HierarchyEventWithInterfaces extends BaseHierarchyEvent
            implements TestEventInterfaceA, TestEventInterfaceB {
        @Override
        public String toString() {
            return "HierarchyEventWithInterfaces{}";
        }
    }

    public static class HierarchyEventWithInterfacesEventAppliedRecorder extends Recorder {
        public void apply(HierarchyEventWithInterfaces event) {
            eventTypeApplied = true;
        }

        @Override
        public String toString() {
            return "HierarchyEventWithInterfacesEventAppliedRecorder{" + "eventTypeApplied=" + eventTypeApplied + '}';
        }
    }

    public static class SubRecorder extends SuperRecorder {
        @Override
        public String toString() {
            return "SubRecorder{" + "eventTypeApplied=" + eventTypeApplied + '}';
        }
    }

    public static class SuperRecorder extends Recorder {
        public void apply(Event event) {
            eventTypeApplied = true;
        }

        @Override
        public String toString() {
            return "SuperRecorder{" + "eventTypeApplied=" + eventTypeApplied + '}';
        }
    }

    public interface TestEventInterfaceA extends Event {}

    public static class TestEventInterfaceAEventAppliedRecorder extends Recorder {
        public void apply(TestEventInterfaceA event) {
            eventTypeApplied = true;
        }

        @Override
        public String toString() {
            return "TestEventInterfaceAEventAppliedRecorder{" + "eventTypeApplied=" + eventTypeApplied + '}';
        }
    }

    public interface TestEventInterfaceB extends Event {}

    public static class TestEventInterfaceBEventAppliedRecorder extends Recorder {
        public void apply(TestEventInterfaceB event) {
            eventTypeApplied = true;
        }

        @Override
        public String toString() {
            return "TestEventInterfaceBEventAppliedRecorder{" + "eventTypeApplied=" + eventTypeApplied + '}';
        }
    }

    private abstract static class Recorder implements EventSourceEntity {
        protected boolean eventTypeApplied = false;

        public boolean eventTypeApplied() {
            return eventTypeApplied;
        }

        @Override
        public String toString() {
            return "Recorder{" + "eventTypeApplied=" + eventTypeApplied + '}';
        }
    }
}
