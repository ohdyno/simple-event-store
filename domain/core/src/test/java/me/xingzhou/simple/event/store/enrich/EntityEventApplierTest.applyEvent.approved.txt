LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='apply {} to entity {}', arguments=[EventRecord[event=TestEvent[id=event-id], details=RecordDetails[streamName=StreamName[value=test-aggregate-id], version=Version[value=0], recordId=RecordId[id=1], insertedOn=[Date1]]], TestAggregate{id='test-aggregate-id', appliedEvents=[TestEvent[id=event-id]]}]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to find the first parameter of the first {}(<? extends {}>) method', arguments=[apply, class me.xingzhou.simple.event.store.events.TestEvent]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to find method {}({})', arguments=[apply, class me.xingzhou.simple.event.store.events.TestEvent]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to invoke method {}({})', arguments=[apply, class me.xingzhou.simple.event.store.events.TestEvent]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='apply {} to entity {}', arguments=[EventRecord[event=TestEvent[id=event-id], details=RecordDetails[streamName=StreamName[value=test-aggregate-id], version=Version[value=0], recordId=RecordId[id=1], insertedOn=[Date1]]], ProjectionRecorder{appliedEvents=[TestEvent[id=event-id]]}]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to find the first parameter of the first {}(<? extends {}>) method', arguments=[apply, class me.xingzhou.simple.event.store.events.TestEvent]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to find method {}({})', arguments=[apply, interface me.xingzhou.simple.event.store.Event]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to invoke method {}({})', arguments=[apply, interface me.xingzhou.simple.event.store.Event]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='apply {} to entity {}', arguments=[EventRecord[event=TestEvent[id=null], details=RecordDetails[streamName=StreamName[value=test-aggregate-id], version=Version[value=0], recordId=RecordId[id=1], insertedOn=[Date2]]], TestAggregate{id='test-aggregate-id', appliedEvents=[TestEvent[id=null]]}]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to find the first parameter of the first {}(<? extends {}>) method', arguments=[apply, class me.xingzhou.simple.event.store.events.TestEvent]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to find method {}({})', arguments=[apply, class me.xingzhou.simple.event.store.events.TestEvent]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to invoke method {}({})', arguments=[apply, class me.xingzhou.simple.event.store.events.TestEvent]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='apply {} to entity {}', arguments=[EventRecord[event=TestEvent[id=event-id], details=RecordDetails[streamName=StreamName[value=test-aggregate-id], version=Version[value=0], recordId=RecordId[id=1], insertedOn=[Date3]]], TestAggregate{id='test-aggregate-id', appliedEvents=[TestEvent[id=event-id], TestEvent[id=event-id]]}]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to find the first parameter of the first {}(<? extends {}>) method', arguments=[apply, class me.xingzhou.simple.event.store.events.TestEvent]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to find method {}({})', arguments=[apply, class me.xingzhou.simple.event.store.events.TestEvent]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to invoke method {}({})', arguments=[apply, class me.xingzhou.simple.event.store.events.TestEvent]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='apply {} to entity {}', arguments=[EventRecord[event=TestEvent[id=event-id], details=RecordDetails[streamName=StreamName[value=test-aggregate-id], version=Version[value=1], recordId=RecordId[id=2], insertedOn=[Date4]]], TestAggregate{id='test-aggregate-id', appliedEvents=[TestEvent[id=event-id], TestEvent[id=event-id]]}]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to find the first parameter of the first {}(<? extends {}>) method', arguments=[apply, class me.xingzhou.simple.event.store.events.TestEvent]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to find method {}({})', arguments=[apply, class me.xingzhou.simple.event.store.events.TestEvent]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to invoke method {}({})', arguments=[apply, class me.xingzhou.simple.event.store.events.TestEvent]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='apply {} to entity {}', arguments=[EventRecord[event=TestEvent[id=event-id], details=RecordDetails[streamName=StreamName[value=test-aggregate-id], version=Version[value=0], recordId=RecordId[id=1], insertedOn=[Date3]]], ProjectionRecorder{appliedEvents=[TestEvent[id=event-id], TestEvent[id=event-id]]}]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to find the first parameter of the first {}(<? extends {}>) method', arguments=[apply, class me.xingzhou.simple.event.store.events.TestEvent]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to find method {}({})', arguments=[apply, interface me.xingzhou.simple.event.store.Event]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to invoke method {}({})', arguments=[apply, interface me.xingzhou.simple.event.store.Event]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='apply {} to entity {}', arguments=[EventRecord[event=TestEvent[id=event-id], details=RecordDetails[streamName=StreamName[value=test-aggregate-id], version=Version[value=1], recordId=RecordId[id=2], insertedOn=[Date4]]], ProjectionRecorder{appliedEvents=[TestEvent[id=event-id], TestEvent[id=event-id]]}]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to find the first parameter of the first {}(<? extends {}>) method', arguments=[apply, class me.xingzhou.simple.event.store.events.TestEvent]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to find method {}({})', arguments=[apply, interface me.xingzhou.simple.event.store.Event]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to invoke method {}({})', arguments=[apply, interface me.xingzhou.simple.event.store.Event]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='apply {} to entity {}', arguments=[EventRecord[event=TestEvent[id=null], details=RecordDetails[streamName=StreamName[value=test-aggregate-id], version=Version[value=0], recordId=RecordId[id=1], insertedOn=[Date5]]], TestAggregate{id='test-aggregate-id', appliedEvents=[TestEvent[id=null]]}]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to find the first parameter of the first {}(<? extends {}>) method', arguments=[apply, class me.xingzhou.simple.event.store.events.TestEvent]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to find method {}({})', arguments=[apply, class me.xingzhou.simple.event.store.events.TestEvent]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to invoke method {}({})', arguments=[apply, class me.xingzhou.simple.event.store.events.TestEvent]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='apply {} to entity {}', arguments=[EventRecord[event=TestEvent[id=null], details=RecordDetails[streamName=StreamName[value=test-aggregate-id], version=Version[value=0], recordId=RecordId[id=1], insertedOn=[Date6]]], TestAggregate{id='test-aggregate-id', appliedEvents=[TestEvent[id=null]]}]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to find the first parameter of the first {}(<? extends {}>) method', arguments=[apply, class me.xingzhou.simple.event.store.events.TestEvent]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to find method {}({})', arguments=[apply, class me.xingzhou.simple.event.store.events.TestEvent]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to invoke method {}({})', arguments=[apply, class me.xingzhou.simple.event.store.events.TestEvent]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='apply {} to entity {}', arguments=[EventRecord[event=TestEvent[id=event-id], details=RecordDetails[streamName=StreamName[value=test-aggregate-id], version=Version[value=0], recordId=RecordId[id=1], insertedOn=[Date7]]], TestAggregate{id='test-aggregate-id', appliedEvents=[TestEvent[id=event-id]]}]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to find the first parameter of the first {}(<? extends {}>) method', arguments=[apply, class me.xingzhou.simple.event.store.events.TestEvent]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to find method {}({})', arguments=[apply, class me.xingzhou.simple.event.store.events.TestEvent]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to invoke method {}({})', arguments=[apply, class me.xingzhou.simple.event.store.events.TestEvent]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='apply {} to entity {}', arguments=[EventRecord[event=TestEvent[id=event-id], details=RecordDetails[streamName=StreamName[value=test-aggregate-id], version=Version[value=0], recordId=RecordId[id=1], insertedOn=[Date7]]], TestAggregate{id='test-aggregate-id', appliedEvents=[TestEvent[id=event-id]]}]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to find the first parameter of the first {}(<? extends {}>) method', arguments=[apply, class me.xingzhou.simple.event.store.events.TestEvent]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to find method {}({})', arguments=[apply, class me.xingzhou.simple.event.store.events.TestEvent]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to invoke method {}({})', arguments=[apply, class me.xingzhou.simple.event.store.events.TestEvent]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='apply {} to entity {}', arguments=[EventRecord[event=HierarchyEventWithInterfaces{}, details=RecordDetails[streamName=StreamName[value=stream-name], version=Version[value=0], recordId=RecordId[id=0], insertedOn=[Date8]]], EventAppliedRecorder{}]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to find the first parameter of the first {}(<? extends {}>) method', arguments=[apply, class me.xingzhou.simple.event.store.enrich.EntityEventApplierTest$HierarchyEventWithInterfaces]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to find method {}({})', arguments=[apply, interface me.xingzhou.simple.event.store.Event]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to invoke method {}({})', arguments=[apply, interface me.xingzhou.simple.event.store.Event]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='apply {} to entity {}', arguments=[EventRecord[event=HierarchyEventWithInterfaces{}, details=RecordDetails[streamName=StreamName[value=stream-name], version=Version[value=0], recordId=RecordId[id=0], insertedOn=[Date8]]], BaseHierarchyEventAppliedRecorder{eventTypeApplied=true}]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to find the first parameter of the first {}(<? extends {}>) method', arguments=[apply, class me.xingzhou.simple.event.store.enrich.EntityEventApplierTest$HierarchyEventWithInterfaces]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to find method {}({})', arguments=[apply, class me.xingzhou.simple.event.store.enrich.EntityEventApplierTest$BaseHierarchyEvent]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to invoke method {}({})', arguments=[apply, class me.xingzhou.simple.event.store.enrich.EntityEventApplierTest$BaseHierarchyEvent]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='apply {} to entity {}', arguments=[EventRecord[event=HierarchyEventWithInterfaces{}, details=RecordDetails[streamName=StreamName[value=stream-name], version=Version[value=0], recordId=RecordId[id=0], insertedOn=[Date8]]], TestEventInterfaceAEventAppliedRecorder{eventTypeApplied=true}]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to find the first parameter of the first {}(<? extends {}>) method', arguments=[apply, class me.xingzhou.simple.event.store.enrich.EntityEventApplierTest$HierarchyEventWithInterfaces]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to find method {}({})', arguments=[apply, interface me.xingzhou.simple.event.store.enrich.EntityEventApplierTest$TestEventInterfaceA]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to invoke method {}({})', arguments=[apply, interface me.xingzhou.simple.event.store.enrich.EntityEventApplierTest$TestEventInterfaceA]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='apply {} to entity {}', arguments=[EventRecord[event=HierarchyEventWithInterfaces{}, details=RecordDetails[streamName=StreamName[value=stream-name], version=Version[value=0], recordId=RecordId[id=0], insertedOn=[Date8]]], TestEventInterfaceBEventAppliedRecorder{eventTypeApplied=true}]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to find the first parameter of the first {}(<? extends {}>) method', arguments=[apply, class me.xingzhou.simple.event.store.enrich.EntityEventApplierTest$HierarchyEventWithInterfaces]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to find method {}({})', arguments=[apply, interface me.xingzhou.simple.event.store.enrich.EntityEventApplierTest$TestEventInterfaceB]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to invoke method {}({})', arguments=[apply, interface me.xingzhou.simple.event.store.enrich.EntityEventApplierTest$TestEventInterfaceB]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='apply {} to entity {}', arguments=[EventRecord[event=HierarchyEventWithInterfaces{}, details=RecordDetails[streamName=StreamName[value=stream-name], version=Version[value=0], recordId=RecordId[id=0], insertedOn=[Date8]]], HierarchyEventWithInterfacesEventAppliedRecorder{eventTypeApplied=true}]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to find the first parameter of the first {}(<? extends {}>) method', arguments=[apply, class me.xingzhou.simple.event.store.enrich.EntityEventApplierTest$HierarchyEventWithInterfaces]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to find method {}({})', arguments=[apply, class me.xingzhou.simple.event.store.enrich.EntityEventApplierTest$HierarchyEventWithInterfaces]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to invoke method {}({})', arguments=[apply, class me.xingzhou.simple.event.store.enrich.EntityEventApplierTest$HierarchyEventWithInterfaces]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='apply {} to entity {}', arguments=[EventRecord[event=HierarchyEventWithInterfaces{}, details=RecordDetails[streamName=StreamName[value=stream-name], version=Version[value=0], recordId=RecordId[id=0], insertedOn=[Date8]]], SubRecorder{eventTypeApplied=true}]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to find the first parameter of the first {}(<? extends {}>) method', arguments=[apply, class me.xingzhou.simple.event.store.enrich.EntityEventApplierTest$HierarchyEventWithInterfaces]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to find method {}({})', arguments=[apply, interface me.xingzhou.simple.event.store.Event]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to invoke method {}({})', arguments=[apply, interface me.xingzhou.simple.event.store.Event]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='apply {} to entity {}', arguments=[EventRecord[event=HierarchyEventWithInterfaces{}, details=RecordDetails[streamName=StreamName[value=stream-name], version=Version[value=0], recordId=RecordId[id=0], insertedOn=[Date8]]], EventWithRecordDetailsRecorder{recordedDetails=RecordDetails[streamName=StreamName[value=stream-name], version=Version[value=0], recordId=RecordId[id=0], insertedOn=[Date8]], eventTypeApplied=true}]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to find the first parameter of the first {}(<? extends {}>) method', arguments=[apply, class me.xingzhou.simple.event.store.enrich.EntityEventApplierTest$HierarchyEventWithInterfaces]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to find method {}({})', arguments=[apply, interface me.xingzhou.simple.event.store.Event]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='unable to find method {}({})', arguments=[apply, interface me.xingzhou.simple.event.store.Event]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to find public method {}({}, {})', arguments=[apply, interface me.xingzhou.simple.event.store.Event, class me.xingzhou.simple.event.store.RecordDetails]}
LoggingEvent{level=DEBUG, mdc={}, markers=[], keyValuePairs=[], throwable=Optional.empty, message='attempting to invoke method {}({}, {})', arguments=[apply, interface me.xingzhou.simple.event.store.Event, class me.xingzhou.simple.event.store.RecordDetails]}
